### ISSUE TITLE
Support for arm64 images

### ISSUE THREAD

**Issue by arminus** [eduardolat/pgbackweb#18]
First of all, great idea and looking forward to the planned features!

Since my always-on box at home is a Pi4, I've tried to test this there by extending your Dockerfile for arm64, see [here](https://github.com/eduardolat/pgbackweb/compare/main...arminus:pgbackweb:main). Cross built the image (on a faster amd64) like so:
```
docker buildx create --name mybuilder --use
docker buildx inspect --bootstrap
docker buildx build --platform linux/arm64 -t pgbackweb:latest --output type=docker,dest=pgbackweb.tar .
```
Then, to bypass a docker registry for now, copied the image to the Pi4 and loaded it with
```
docker load -i pgbackweb.tar
```

Works like a charm.

In case you're interested, I'd create a PR, maybe I also have time to checkout how to set this up in a github build pipeline for arm64 but I haven't done this before so no promises ;-)

**Comment by eduardolat** [comment]
Great! It can be very useful, just make sure your PR doesn't add too much complexity to the project, as it's important to keep it as simple as possible.

Thanks for considering contributing to the project.

**Comment by arminus** [comment]
No complexity, it's just platform specific wgets in your Dockerfile, see https://github.com/eduardolat/pgbackweb/pull/21

I see no github pipeline for how you build and publish your Docker image, so I can't automate that I'm afraid.

**Comment by ykrasik** [comment]
+1 on this, please

**Comment by ThisIsBenny** [comment]
+1

**Comment by eduardolat** [comment]
Thanks @arminus for your great help.

Since you opened your pr, there have been many changes to the Dockerfiles, the base image has been changed, dependencies have been added, and many other things.

However, I have taken the changes proposed in your pr #21 and adapted them to the new Dockerfiles.

I have also included a script that checks that the dependencies are working correctly and I have added it to the github actions tests workflow.

All that is left is to add a github action to automatically publish the image in each release.

Everything is working perfectly and this will be published in the next release, thanks for your help and thanks to everyone for the interest in the project 🍻

![image](https://github.com/user-attachments/assets/92aaf075-996c-45b2-aaa6-d8353531cdcc)

**Comment by eduardolat** [comment]
This has been published in the v0.3.0 release https://github.com/eduardolat/pgbackweb/releases/tag/0.3.0

**Comment by ThisIsBenny** [comment]
I got the error:
`exec user process caused: exec format error`
Which usually happens when it is not an ARM64 image.
Is the image working for some one?

**Comment by eduardolat** [comment]
hey @ThisIsBenny, can you try to pull the latest version with this command?

`docker pull --platform=linux/arm64 eduardolat/pgbackweb:0.3.0`

And then explicitly set in your compose file the version of your image to
```
services:
  pgbackweb:
    image: eduardolat/pgbackweb:latest:0.3.0

## rest of your config
```

**Comment by ThisIsBenny** [comment]
I have used it with my Raspberry PI Kubernetes Cluster, why i'm not able to specify the platform manually. But I have used the 0.3.0 Tag and also tried with the SHA hash of the Arm image like 0.3.0@sha....

**Comment by eduardolat** [comment]
Can you please try (just for test purposes) with the docker compose in the README.md?

**Comment by arminus** [comment]
```
docker pull --platform=linux/arm64 eduardolat/pgbackweb:0.3.0
````
then compose
```
image: eduardolat/pgbackweb:0.3.0
```
(note: 2 tags are illegal syntax)

**does not work, still getting exec format error.**

I assume the Manifest for the images is wrong or missing:
```
docker manifest inspect eduardolat/pgbackweb:0.3.0
no such manifest: docker.io/eduardolat/pgbackweb:0.3.0
```

Normally (with all other images I run on my Pi4), I don't have to explicitly set the platform when pulling, docker figures this out on its own if the images are configured properly (but I haven't built multiplatform images myself yet, so I can only guess here)

**Comment by electric-m** [comment]
> 
> **does not work, still getting exec format error.**
> 

Same issue for me here an a Hetzner ARM VPS.
Explicitly defined pulled an d defined the 0.30. tag.
```
~/pgbackweb$ docker image inspect eduardolat/pgbackweb:0.3.0 | grep Architecture
        "Architecture": "arm64",
```

**Comment by eduardolat** [comment]
You are totally right! the fix is on the way on the develop branch, will be included in next release

**Comment by electric-m** [comment]
Ok, Great!

**Comment by electric-m** [comment]
Can I already use ARM64 by building with the Dockerfile from the dev branch, or do I have to wait for 0.3.x?

**Comment by danielehrhardt** [comment]
Some Updates here?

**Comment by eduardolat** [comment]
Guys, I'm pleased to inform you that this issue has been resolved as of v0.4.0 which you can start using now.

I've improved the way images are built and tested thanks to the new ARM runners on github actions.

I've also tested the resulting images myself on an ARM VPS with an Ampere processor.

I hope you like it.

**Comment by electric-m** [comment]
> Guys, I'm pleased to inform you that this issue has been resolved as of v0.4.0 which you can start using now.
> 
> I've improved the way images are built and tested thanks to the new ARM runners on github actions.
> 
> I've also tested the resulting images myself on an ARM VPS with an Ampere processor.
> 
> I hope you like it.

Thank you so much, it is awesome!
